src/vaadin-menu-bar-buttons-mixin.html

```diff
@@ -169,6 +206,55 @@
       }
     }
 
+    render() {
+      this.__renderButtons(this.items);
+    }
+
+    __renderButtons(items = []) {
+      const container = this._container;
+      const ellipsis = container.querySelector('[part="ellipsis-button"]');
+
+      while (container.children.length > 1) {
+        container.removeChild(container.firstElementChild);
+      }
+
+      items.forEach(item => {
+        const button = document.createElement('vaadin-menu-bar-button');
+        button.item = item;
+
+        const itemComponent = item.component;
+        if (itemComponent) {
+          let component;
+          const isElement = itemComponent instanceof HTMLElement;
+          // use existing item component, if any
+          if (isElement && itemComponent.localName === 'vaadin-context-menu-item') {
+            component = itemComponent;
+          } else {
+            component = document.createElement('vaadin-context-menu-item');
+            component.appendChild(isElement ? itemComponent : document.createElement(itemComponent));
+          }
+          if (item.text) {
+            const node = component.firstChild || component;
+            node.textContent = item.text;
+          }
+          // save item for ellipsis menu
+          component.item = item;
+          button.item.component = component;
```

Comment:
So this changes each `component` reference in the user-provided `items`-array to the generated `vaadin-context-menu-item`. I think it's a bad idea to make this kind of changes directly to the data structure provided by the user, even though in this case it's rare to cause any issues.

Maybe make a copy of `item` to not mutate user's data?

Reply:
Updated the code to create a copy of item and not mutate the `items` array.